### ISSUE TITLE
HFW 4.90.1 (Hybrid Firmware) hard drive exfat not showing up prepISO

### ISSUE THREAD

**Issue by syfer1987** [aldostools/webMAN-MOD#1008]
hi on HFW 4.90.1 (Hybrid Firmware) and on webMAN MOD 1.47.45 but the hard drive 10tb and 12tb both don't show up after running  prepISO. Game folder is PS3ISO. Note I installed full version. Need Help I'm just normal user so please don't use molding Gargan.

**Comment by aldostools** [comment]
To load webMAN MOD you need to have PS3HEN enabled.

Games in PS3ISO folder must be in ISO format.

I recommend you that install IRISMAN to verify if the File Manager can recognize your external drives. prepISO uses the same NTFS and exFAT libraries to read the disk. 

PS3 normally only recognizes MBR external disks of up to 2TB. To recognize the 10tb and 12tb in the external HDD, they need to be formatted in NTFS or exFAT with large clusters of 16KB or 64KB and the disk partition must be the first  one in MBR (Master Boot Record). GPT is supposed to be supported, but it has not been verified.

For further assistance, please ask in PSX-PLACE forum.

**Comment by syfer1987** [comment]
Ok i will try the cluster size so for exFAT it would be 16KB or 64KB but what about ntfs what cluster size does ntfs need to be?

**Comment by aldostools** [comment]
Same. The default size cluster size for NTFS is 4KB

**Comment by RinMaru** [comment]
64k clusters make the drive not show up

**Comment by RinMaru** [comment]
Do you have any other partitions on the exfat drive? Exfat wont work unless its the only partition on your drive

**Comment by aldostools** [comment]
I found in exFAT lib that there is a parameter that allows multi-partitions. But I never have used that feature.
I don't know if it requires extra code to make it work, of if it only needs the value changed.

![Image](https://github.com/user-attachments/assets/decfc129-23e0-464f-b406-7828b919eee8)

You can test this build of prepISO. I compiled it with the setting changed from the default 0 to 1.
[prepISO_TEST_EXFAT_MULTI_PARTITION.zip](https://github.com/user-attachments/files/19149514/prepISO_TEST_EXFAT_MULTI_PARTITION.zip)

**Comment by RinMaru** [comment]
> I found in exFAT lib that there is a parameter that allows multi-partitions. But I never have used that feature.
> I don't know if it requires extra code to make it work, of if it only needs the value changed.
> 
> ![Image](https://github.com/user-attachments/assets/decfc129-23e0-464f-b406-7828b919eee8)
> 
> You can test this build of prepISO. I compiled it with the setting changed from the default 0 to 1.
> [prepISO_TEST_EXFAT_MULTI_PARTITION.zip](https://github.com/user-attachments/files/19149514/prepISO_TEST_EXFAT_MULTI_PARTITION.zip)

Ill try it at some point. It takes 9hrs to transfer 1.2TB of backups. I did notice PrepISO doesnt fetch boxart for games on exfat or lable that by param. Also it doesnt create a cache for isos it seems. It can scan and get them to show up but mounting them causes a freeze in functions. Only webman full can scan and create cache for exFAT so they mount

**Comment by aldostools** [comment]
To speed up the tests you can copy only a single ISO after format, and use dummy files to simulate more files.

In regards to the issue with box arts, it's weird that it isn't copying the covers. exFAT performs 2 loops per directory: the first copy the images (.png or .jpg) and the second loop create the cache files for ISOs and copy .key or .dkey.

If the file size of box art is greater than 4MB, it will be ignored.

**Comment by RinMaru** [comment]
Yea its just creating entries but no SFO files. You have to get webMAN to scan them. WebMAN alone doesnt seem to see games on exFAT when you scan with it. Only after PrepISO has created an entry will it see it
Gonna give NTFS 64K a try again maybe its picky like exFat

**Comment by aldostools** [comment]
Yes, webMAN MOD only has native support for NTFS. ExFAT lib is not compatible with the PS3 SDK required to build VSH plugins.
prepISO is required to scan the games stored in exFAT.

If the SFO files are already extracted in exFAT, they should be copied like the cover images. The internal function in prepISO that allows to extract file from ISO files is only compatible with NTFS and ext2/3/4.

**Comment by RinMaru** [comment]
that makes sense. seems exFAT is pretty new then. the 64K cluster size did make GTA4 initial install half the time it took on 32k NTFS. 
and no i dont have the sfo extracted. i do have my backups decrypted though.
Update NTFS 64k Cluster size is just completely invisible to anything on ps3. webMAN,SFM,and PrepISO don't see it. PrepISO makes the drive blink so its being read but nothing is returned. 64K is possible on FAT32 and exFAT though

Think you can try 64K NTFS on your end? I think this should be noted in the Wiki if its definitive

**Comment by aldostools** [comment]
I am unable to test at this time. My PS3 is faulty and I am unable to devote time to this problem due to other obligations.

**Comment by RinMaru** [comment]
> I am unable to test at this time. My PS3 is faulty and I am unable to devote time to this problem due to other obligations.

Thats fine. Good news is 128k cluster size works on exFAT but the difference in speed isn't really there if only by a few ms. or Slower.

**Comment by aldostools** [comment]
It's important to note that if you're testing with the test build prepISO_TEST_EXFAT_MULTI_PARTITION, that version is different to the public release.

**Comment by RinMaru** [comment]
> It's important to note that if you're testing with the test build prepISO_TEST_EXFAT_MULTI_PARTITION, that version is different to the public release.

I havent tested it yet. is there more going on with it outside partition detection? Gonna install it.

**Comment by aldostools** [comment]
> I havent tested it yet. is there more going on with it outside partition detection? Gonna install it.

No. Just recompiled prepISO with the parameter for the exFAT multi-partition detection changed from 0 to 1.

**Comment by RinMaru** [comment]
> > I havent tested it yet. is there more going on with it outside partition detection? Gonna install it.
> 
> No. Just recompiled prepISO with the parameter for the exFAT multi-partition detection changed from 0 to 1.

Itll have to wait till tomorrow. cant manage any partitions due to AOMEI Partition Assistant. It's locking me out of messing with drives in Windows while it moves data around since I shrank down my backup partition.

But it would be really helpful if someone could test the 64k NTFS. I updated the post on PSX-Place with a request for anyone to take up the task. really curious if its a limitation of not full CFW PS3s

**Comment by RinMaru** [comment]
> > I havent tested it yet. is there more going on with it outside partition detection? Gonna install it.
> 
> No. Just recompiled prepISO with the parameter for the exFAT multi-partition detection changed from 0 to 1.

LIAR you made it able to read the sfo too lol this is GREAT. if that wasn't intentional then i guess we both surprised?
WAIT maybe that was just me its remembering me mounting the games in the browser previously and didnt give thumbnails till restart

**Comment by aldostools** [comment]
I don't lie... I take my tasks very seriously. Maybe you mounted the game and webMAN MOD copied the SFO to wmtmp.

prepISO does not have the ability to read the SFO from the ISO on exFAT, unless you put it externally in the same folder of the ISO.

**Comment by aldostools** [comment]
By the way. If you include the title id in the file name of the ISO, the cover is show from the covers database. Therefore is not necessary to read the PARAM.SFO except for the game title.

**Comment by RinMaru** [comment]
> I don't lie... I take my tasks very seriously. Maybe you mounted the game and webMAN MOD copied the SFO to wmtmp.

Yea I know ^^. that was indeed the case. I wish it could read it. oh example "Dead Space [ID].iso"

**Comment by aldostools** [comment]
It should be able to read the ID from the filename. example: "Dead Space [BLUS30177].iso"

You need to have installed the [Covers pack](https://github.com/aldostools/resources/releases/1.0)

**Comment by RinMaru** [comment]
> It should be able to read the ID from the filename. example: "Dead Space [BLUS30177].iso"
> 
> You need to have installed the [Covers pack](https://github.com/aldostools/resources/releases/1.0)

I dont have that installed but PrepISO seems to grab the covers if the title ID is present. made sure to clear wmmtmp folder and restart before scanning again

**Comment by aldostools** [comment]
Yes, it does if you have selected ONLINE COVERS. The covers pack is needed for offline display.

**Comment by RinMaru** [comment]
> Yes, it does if you have selected ONLINE COVERS. The covers pack is needed for offline display.

Just got done renaming 75 games and formatting my 2TB drive to-

- Partition 1 1.5TB exFAT 64k Cluster 
- Partition 2 220GB FAT32 64k Cluster. 

Transfer will take awhile but ill report back tomorrow if PrepISO detects both Partitions

**Comment by RinMaru** [comment]
@aldostools nope it still only reads the fat32 partition
